### DISCUSSION TITLE
Where is chat history? (Feature ideas)

### DISCUSSION THREAD

**Discussion by DynDux** [rjmacarthy/twinny#187] [Q&A]
In the plugin description you write: "Chat history preserved per workspace"

What does that mean? I'm looking for a function to see previous chats and maybe continue with one of them. Like within the "Continue" Plugin:

![twinny-history](https://github.com/rjmacarthy/twinny/assets/76206890/3cf4518c-daeb-47f9-8ab3-a6856cef1968)

Does this feature exists already and I just didn't find it?

Some small feature ideas:
- resubmit a question with new selection context
- add a button to quickly copy a question that you want to ask again

Really love this Plugin! Thanks so much for sharing!

**Comment by rjmacarthy** [comment]
Thanks for the interest. It means that the chat is saved on your workspace(project) until you clear it. To save all conversations would require more work, definitely possible but just needs a bit of time/effort.

**Comment by DynDux** [comment]
Ah - ok! That clears things up. Thanks!
I will patiently wait - maybe in a future version ;-)

**Reply by rjmacarthy** [reply]
It's ready now 🙂

**Reply by DynDux** [reply]
What? 😲 
Whoa man, you are awesome! 😀👍

**Comment by rjmacarthy** [comment]
https://github.com/rjmacarthy/twinny/releases/tag/v3.11.9 

Please update and see the release notes.

**Reply by DynDux** [reply]
Already done that - works flawless!
I'm impressed how fast and committed you are. Thanks so much!
